test/mlj_model_interface.jl

```diff
@@ -37,6 +37,48 @@ end
     end
 end
 
+@testset "regularization" begin
+    rng = StableRNG(123)
+    nobservations = 12
+    Xuser = rand(Float32, nobservations, 3)
+    yuser = rand(Float32, nobservations)
+    alpha = rand(rng)
+    lambda = rand(rng)
+    optimiser = Optimisers.Momentum()
+    builder = MLJFlux.Linear()
+    epochs = 1 # don't change this
+    opts = (; alpha, lambda, optimiser, builder, epochs)
+
+    for batch_size in [1, 2, 3]
+
+        # (1) train using weight/sign decay, as implemented in MLJFlux:
+        model = MLJFlux.NeuralNetworkRegressor(; batch_size, rng=StableRNG(123), opts...);
+        mach = machine(model, Xuser, yuser);
+        fit!(mach, verbosity=0);
+        w1 = Optimisers.trainables(fitted_params(mach).chain)
+
+        # (2) manually train for one epoch explicitly adding a loss penalty:
+        chain = MLJFlux.build(builder, StableRNG(123), 3, 1);
+        penalty = Penalizer(lambda, alpha); # defined in test_utils.jl
+        X, y = MLJFlux.collate(model, Xuser, yuser);
+        loss = model.loss;
+        n_batches = div(nobservations, batch_size)
+        optimiser_state = Optimisers.setup(optimiser, chain);
+        for i in 1:n_batches
+            batch_loss, gs = Flux.withgradient(chain) do m
+                yhat = m(X[i])
+                loss(yhat, y[i]) + sum(penalty, Optimisers.trainables(m))/n_batches
```

Comment:
@pat-alt Just FYI: Optimisers.jl added `trainables` which offers this solution to refactor along your original line. It superficially seems like `Flux.params`, but doesn't seem to suffer the same problems we were seeing (if you can remember back that far :wink:)

Reply:
Thanks! 

So from a user-persective, I'm currently unclear about a few things (perhaps I've just not looked carefully enough!):

1. Is the `regularised_optimiser` method user-facing? Where/when is it used here in this specific unit test?
2. Would it make sense to ship the `Penalizer`?